server.go

```diff
@@ -138,8 +141,12 @@ func NewServer(options ...func(*Server)) *Server {
 		option(s)
 	}
 
+	if s.Server.Addr == "" {
+		WithAddr("localhost:9999")(s)
+	}
+
 	s.OpenApiSpec.Servers = append(s.OpenApiSpec.Servers, &openapi3.Server{
-		URL:         "http://" + s.Addr,
+		URL:         fmt.Sprintf("%s://%s", s.proto(), s.Server.Addr),
```

Comment:
nice

Reply:
I believe this would always be just http at this point except in the case where a Listener is passed. The current behavior is that we don’t know until the user calls RunTLS which is fine. 

I think we’d need even an even larger refactor to address this well. This falls out of scope of this PR

Reply:
Hmmm. Actually probably the proper places is at the end of setup(). 

I could be wrong it’s late 😅

Reply:
I agree, I thought it could be a problem too.

So does it have to be merged like that? With somehow a bug?

Or the refactoring should gf done before? Either here or in another PR?

Reply:
Doing it in setup now is fine.